### ISSUE TITLE
Feature Request: annotating rule for spec.matches.labelSelector

### ISSUE THREAD

**Issue by dulltz** [plexsystems/konstraint#93]
I would like to use Constraint's [`spec.match.labelSelector`](https://github.com/open-policy-agent/gatekeeper/tree/v3.2.0#constraints),
but konstraint seems to only supports `spec.match.kinds` now. ([ref](https://github.com/plexsystems/konstraint/blob/771677eac46ba1e870af5cdebfd6018ee8c489a7/internal/rego/matchers.go#L29-L38))

Also, as described in https://github.com/plexsystems/konstraint/issues/56, rerunning of `konstraint create` erases existing `spec.match.labelSelector`.

We should support an annotation rule for generating `spec.matches.labelSelector`.

**Comment by dulltz** [comment]
I want to add annotation rules like the following example.

Supporting all representations of [labelSelector](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.19/#labelselector-v1-meta) with annotation rules would be complicated, so I think it is simpler to support only matchLabels.

```rego
# @matchlabels app=a,team=b
package foo

import data.lib.core
...
```
↓
```yaml
apiVersion: constraints.gatekeeper.sh/v1beta1
kind: Foo
metadata:
  name: foo
spec:
  match:
    labelSelector:
      matchLabels:
        app: a
        team: b
```

**Comment by jalseth** [comment]
I think this approach makes sense. We will start with `matchLabels` and we can look into adding support for `matchExpressions` if that is requested in the future. If you have the time, we'd welcome a PR to add this!

**Comment by dulltz** [comment]
@jalseth
I created the PR https://github.com/plexsystems/konstraint/pull/94. May I ask for your review?

**Comment by jalseth** [comment]
Implemented in #94
